### PR TITLE
feat(condo): DOMA-8365 changed card styles, added checkbox card

### PR DESCRIPTION
**Pull request by nomerdvadcatpyat**
Default card:
изображение

Card active state:
изображение

Button card with full content:
изображение

Checkbox card with full content:
изображение

Checkbox card active state:
изображение

### CODE REVIEW

**Review by SavelevMatthew — COMMENTED** [review]
1. Checkbox hover behaviour is independent from card one. 
2. Checkbox card itself have no hover behaviour. 
3. WTF is black card about? I thought we're building light interfaces

**Comment by nomerdvadcatpyat** [comment]
> 1. Checkbox hover behaviour is independent from card one.

Fixed. Added `pointer-events: none` to inner checkbox

>  2. Checkbox card itself have no hover behaviour.

Fixed. Added default `hoverable` to `CheckboxCard`, omit `hoverable` from `CheckboxCard` props

>  3. WTF is black card about? I thought we're building light interfaces

It's [new active state of card](https://www.figma.com/file/Nbkug4IRM33sEyDuN0yO1C/UI-Kit-(web)?type=design&node-id=8732-4676&mode=design&t=r5VgJdYuQ6Akn0lP-0)
For example, [active step in onboarding](https://www.figma.com/file/NnpWfYUo4Z8LGV3gHXRnHe/%D0%9E%D0%BD%D0%B1%D0%BE%D1%80%D0%B4%D0%B8%D0%BD%D0%B3?type=design&node-id=2301-17924&mode=design&t=W4Y1GoxovV6bgAqu-0)

**Comment by Alllex202** [comment]
1) Should it be possible to add a button on a card with a checkbox?
2) Should the state of checkbox change when button is clicked?

https://github.com/open-condo-software/condo/assets/56914444/677d1d28-4d03-4717-aedc-4907a17ae063

**Comment by Alllex202** [comment]
Something's not working here

Снимок экрана 2024-02-13 в 23 18 04
Снимок экрана 2024-02-13 в 23 17 56

**Comment by Alllex202** [comment]
Perhaps we should merge these stories under one section “Card”

Снимок экрана 2024-02-13 в 23 38 04

File: packages/ui/src/components/Card/_utils/cardBody.less
[the remark this answers is not in the document]
Comment (Alllex202) [reply packages/ui/src/components/Card/_utils/cardBody.less:1]: maybe such a structure? Like in `Dropdown`

Card
-- body
---- .tsx, .less
-- header
---- .tsx, .less
-- button
---- .tsx, .less
-- checkbox
---- .tsx, .less

File: packages/ui/src/components/Card/_utils/utils.tsx
Comment (Alllex202) [inline packages/ui/src/components/Card/_utils/utils.tsx:1]: `.../_utils/utils...`
Why the double name? It is already clear that this is a folder with utilities

File: packages/ui/src/components/Card/_utils/utils.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    AfterIcon?: React.FC<IconProps>
+}
+
+export const renderLink = (linkProps: CardLinkType) => {
+    const { PreIcon, AfterIcon, href, label } = linkProps
+
+    return (
+        <Space size={8} direction='horizontal' align='center'>
+            {PreIcon && <PreIcon size='small' />}
+            <Typography.Link href={href}>{label}</Typography.Link>
```
Comment (Alllex202) [inline packages/ui/src/components/Card/_utils/utils.tsx:22]: Will this link be able to work as an SPA link (without reloading the page)? Or should it be like a default link?

File: packages/ui/src/components/Card/cardButton.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+import './cardCheckbox.less'
+import { CardBody, CardBodyProps } from './_utils/cardBody'
+import { CardHeader, CardHeaderProps } from './_utils/cardHeader'
+import { CARD_CLASS_PREFIX } from './_utils/constants'
+import { Card, CardProps } from './card'
+
+
+export type CardButtonProps = Pick<CardProps, 'accent' | 'disabled'> & {
+    header: CardHeaderProps
```
Comment (Alllex202) [inline packages/ui/src/components/Card/cardButton.tsx:12]: Why can't we set a body without a header?

Comment (nomerdvadcatpyat) [reply packages/ui/src/components/Card/cardButton.tsx:12]: Made header optional

File: packages/ui/src/components/Card/_utils/utils.tsx (hunk 1 above)
Comment (nomerdvadcatpyat) [reply packages/ui/src/components/Card/_utils/utils.tsx:22]: By default it's be like default link.
Added `LinkWrapper` prop to make it possible to be spa link (wrap in next/link from outside)

**Comment by nomerdvadcatpyat** [comment]
> 1. Should it be possible to add a button on a card with a checkbox?
> 
> 2. Should the state of checkbox change when button is clicked?
> 
> 
> 2024-02-13.23.32.16.mov

1. Yes
2. No

File: packages/ui/src/components/Emoji/Emoji.tsx
[the remark this answers is not in the document]
Comment (nomerdvadcatpyat) [reply packages/ui/src/components/Emoji/Emoji.tsx:1]: I don’t think it’s necessary yet

File: packages/ui/src/components/Card/checkbox/cardCheckbox.tsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        ...rest
+    } = props
+
+    const [checked, setChecked] = useState<boolean>()
+
+    const handleClick = useCallback((event: React.MouseEvent<HTMLDivElement>) => {
+        setChecked(prev => !prev)
+
+        if (props.onClick) {
+            props.onClick(event)
```
Comment (Alllex202) [inline packages/ui/src/components/Card/checkbox/cardCheckbox.tsx:32]: I think here we should return the value of the checkbox

File: packages/ui/src/components/Card/checkbox/cardCheckbox.tsx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+const CardCheckbox = React.forwardRef<HTMLDivElement, CardCheckboxProps>((props, ref) => {
+    const {
+        header,
+        body,
+        ...rest
+    } = props
+
+    const [checked, setChecked] = useState<boolean>()
```
Comment (Alllex202) [inline packages/ui/src/components/Card/checkbox/cardCheckbox.tsx:26]: what about default value?

File: packages/ui/src/components/Card/checkbox/cardCheckbox.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    const handleClick = useCallback((event: React.MouseEvent<HTMLDivElement>) => {
+        setChecked(prev => !prev)
+
+        if (props.onClick) {
+            props.onClick(event)
+        }
+    }, [props])
```
Comment (Alllex202) [inline packages/ui/src/components/Card/checkbox/cardCheckbox.tsx:34]: We don't need to track all props

File: packages/ui/src/stories/Card/CardCheckbox.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
+import { Meta, Story } from '@storybook/react'
+import {  pickBy } from 'lodash'
```
Comment (Alllex202) [inline packages/ui/src/stories/Card/CardCheckbox.stories.tsx:2]:
```suggestion
import { pickBy } from 'lodash'
```

File: packages/ui/src/stories/ProgressIndicator.stories.tsx
Comment (Alllex202) [inline packages/ui/src/stories/ProgressIndicator.stories.tsx:1]: I think you should send this component for design review

**Comment by Alllex202** [comment]
Снимок экрана 2024-02-15 в 01 33 04

**Comment by Alllex202** [comment]
Should the picture shrink so much on the sides?

Снимок экрана 2024-02-15 в 01 33 04

**Comment by Alllex202** [comment]
Links also change the state of the checkbox. I don't think it should be like that

https://github.com/open-condo-software/condo/assets/56914444/243dde0a-30a9-4053-a693-a53676cc261d

**Comment by Alllex202** [comment]
Something is wrong with picture

https://github.com/open-condo-software/condo/assets/56914444/f4ad6238-929a-4ce4-88d9-44f4d34741d1

**Review by Alllex202 — APPROVED** [review]

**Pull request merged**